From: "Yan, Zheng" <zyan@redhat.com>
Date: Tue, 23 May 2017 17:03:12 +0800
Subject: ceph: remove useless page->mapping check in writepage_nounlock()
Git-commit: f2b0c45f09796f87723a1225c919035457f72b7a
Patch-mainline: v4.13-rc1
References: bsc#1048228

Callers of writepage_nounlock() have already ensured non-null
page->mapping.

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/addr.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/fs/ceph/addr.c
+++ b/fs/ceph/addr.c
@@ -533,10 +533,6 @@ static int writepage_nounlock(struct pag
 
 	dout("writepage %p idx %lu\n", page, page->index);
 
-	if (!page->mapping || !page->mapping->host) {
-		dout("writepage %p - no mapping\n", page);
-		return -EFAULT;
-	}
 	inode = page->mapping->host;
 	ci = ceph_inode(inode);
 	fsc = ceph_inode_to_client(inode);
